posts/python-in-production-part-5-of-5.md

```diff
@@ -0,0 +1,250 @@
+---
+authors:
+- Lucas Ward
+tags:
+- python
+- api
+- devops
+- jenkins
+- ansible
+title: "Python in Production Part 4 of 5"
+image: 
+---
+
+Welcome back to the Python in Production series.  In the [fourth part](https://templinkplaceholder) of the series, we learned how to run our program as a service on a linux machine using systemd.  In the part of the series, we will learn how to automate our build process using Jenkins.
+
+In order not to overwhelm, I am going to assume that you already have a Remote Server up and Running with Docker installed, or at the very least, Docker installed locally.  If you do not have Docker running, you can find the installation instructions [here](https://docs.docker.com/engine/install/). I am also going to assume that you have kept your code up to date in the github repository that we set up in part 1.  What we will learn how to do:
+* Write a Jenkinsfile
+* Connect Github to Jenkins
+* Create a Build Pipeline
+
+# Create a Jenkinsfile
+
+When we attach our github repository to Jenkins, we will be able to tell it to *look* in our repository for instructions on how to build our code.  Those instructions live in a file named `Jenkinsfile`.  In the top level project directory `sample-python-project`, create a file called `Jenkinsfile`.
+```bash
+touch Jenkinsfile
+```
+
+Now we will populate it with our pipeline build instructions.  There are lots of different options and formats here, but we are going to keep it *as simple* as possible.  Open up the Jenkinsfile in your favorite text editor and write the following code.
+```groovy
+pipeline {
+    agent { label "built-in" }
+    stages {
+        stage('build') {
+            steps {
+                sh "bash build.sh"
+            }
+        }
+    }
+}
+
+```
+
+The code here is fairly straightforward to read.  If you were to find this file in the interwebs and attempted to read it you might come to these conclusions.  This is a pipeline file for Jenkins.  It consists of one stage named `build`, that runs one step.  The step is simply a shell command to run our `build.sh` file.  Remember the `build.sh` file we created in part 3, it will now be used inside of our Jenkins runner.  A Jenkins runner is just a linux vm.
+
+# Install Jenkins Plugins
+
+## Github Plugin
```

Comment:
Make sure to standardize how you use Github. Sometimes you use Github, others GitHub